Automattic/wp-calypso
# Deprecate Theme Switch modal

Related to https://github.com/Automattic/wp-calypso/issues/69687
Site Onboarding Flows Homepage Setup > Switching to another theme PCYsg-Ln8-p2
Theme Switch User Research pekYwv-uV-p2
Theme Experience Walkthrough > Issues for preserving templates pekYwv-BQ-p2 
Persist template content on theme switch pbxlJb-2X4-p2

## Proposed Changes

* Remove the theme switch modal
* Pass `dont_change_homepage: true` in the request `POST /themes/mine`

## Description

In this PR, we always pass `dont_change_homepage: true` in the request for a theme switch, which is the same as choosing the "Switch theme, preserving my homepage content" option in the modal. 

We are forcing users to preserve their homepage content on theme switches, but we might have exceptions when testing.

**Where is the logic for `POST /themes/mine`?**
- Simple sites fbhepr%2Skers%2Sjcpbz%2Schoyvp.ncv%2Serfg%2Sjcpbz%2Qwfba%2Qraqcbvagf%2Spynff.jcpbz%2Qwfba%2Qncv%2Qnpgvir%2Qgurzr%2Qraqcbvag.cuc%3Se%3Q1p463405%2330-og
- Atomic sites fbhepr%2Skers%2Swrgcnpx%2Scebwrpgf%2Scyhtvaf%2Swrgcnpx%2Swfba%2Qraqcbvagf%2Swrgcnpx%2Spynff.wrgcnpx%2Qwfba%2Qncv%2Qgurzrf%2Qnpgvir%2Qraqcbvag.cuc%3Se%3Q3n3p4nqp%2336-og

### TODO
 - Add a modal to explain exactly what will happen, including what will happen to their "in-template" content of their current theme and a mention to their reading settings.
 - Update the field guide page: Site Onboarding Flows Homepage Setup > Switching to another theme PCYsg-Ln8-p2

### Known issues
- https://github.com/WordPress/gutenberg/issues/25071
- https://github.com/WordPress/gutenberg/issues/47420
- https://github.com/WordPress/gutenberg/issues/31954

## Testing Instructions

**The expected result after a theme switch is that the content on the homepage (front of the site), excluding template parts (e.g. header/footer), must be preserved.** 

Currently when you activate a new theme, any of the old theme templates that you have customised are retained. For this reason, even though template parts of the homepage template are not preserved after switching to another theme, they will be back if we switch back to the old theme.

The homepage outcome after a theme switch depends on the following:

- the previous homepage templates edited before the switch
- the site reading settings before the switch
- the theme you switch from
- the theme you switch to

### Test flows

**Flow 1 "Skip"**
  - Create a new site
  - Skip to dashboard 
  - Switch theme from `Appearances -> Themes` multiple times

**Flow 2 "Pick a design"** 
  - Create a new site
  - Pick a theme from the design picker
  - Switch theme from `Appearances -> Themes` multiple times

**Flow 3 "Assembler"** 
  - Create a new site
  - Click "Start designing" from the bottom of the design picker
    - Flow 3.1 
      - Add patterns
    - Flow 3.2 
      - Don't add patterns
    - Continue
  - Switch theme from `Appearances -> Themes` multiple times

### Test buttons

From the theme detail, click "Activate this design" 

<img width="653" alt="Screenshot 2566-02-24 at 18 17 36" src="https://user-images.githubusercontent.com/1881481/221166294-23cffa34-940c-4497-b068-75a76890fb15.png">

From the theme showcase, open the three dots menu and click "Activate"

<img width="520" alt="Screenshot 2566-02-24 at 18 17 21" src="https://user-images.githubusercontent.com/1881481/221166471-2ffde9a1-a080-4ba6-8272-4ae410b78df9.png">

### Test themes switch by theme type
- Classic theme => Classic theme
- Classic theme => Block theme
- Block theme => Block theme
- Block theme => Classic theme

### Test themes switch by homepage type
#### Classic themes
Classic themes with **posts** as homepage
  - e.g.: Dyad 2, Independent Publisher 2, Ixion 

Classic themes that uses a **page** as homepage 
 - e.g.: Alves, Twenty Seventeen

#### Block themes
Block themes with **posts** as homepage 
  - whose homepage template is "Index"
    - e.g.: Zoologist, Geologist, Rainfall, Masu
  - whose homepage template is "Home"
    - e.g.: Pendant / Archeo / Skatepark

Block themes that uses a **page** as homepage
  - whose homepage template is "Header and Footer Only"
    - e.g.: Appleton / Dorna / Russell
  - whose homepage template is "Footer Only"
    - e.g.: Ames

### Test sites

From `wordpress.com/themes/{ site slug }`
 - Simple (free)
 - Atomic (business)
 - Jetpack (jurassic.ninja + connection)


## Pre-merge Checklist
